### ISSUE TITLE
Dos build of Vi uses up 100% of CPU

### ISSUE THREAD

**Issue by osen** [open-watcom/open-watcom-v2#353]
inside "open-watcom-v2/bld/vi/dos/dossys.c" the function "MyDelay" does a tightloop spending the cycles.
Is there any way you guys could add some sort of delay() instead so that the HLT instruction can be used.

I am a massive fan of Watcom Vi on DOS and this would certainly save me a warm lap :)

**Comment by jmalak** [comment]
I checked OW 2.0 version of VI and it was completly broken
I fixed it and add calling DOS idle function to keyboard/wait loop that new version should not exhibit this problem
Please, [download latest build of OW 2.0 VI DOS version](https://github.com/open-watcom/travis-ci-ow-builds/blob/master/binw/vi.exe)  and check.

**Comment by osen** [comment]
Firstly, thanks for looking into this.

I downloaded a ran the build of vi.exe that you linked to. I replaced my copy of C:\WATCOM\binw\vi.exe with it and ran it.
Unfortunately my processor usage for Qemu rose from ~20% to ~80% suggesting it is still spinning loads of cycles somewhere.
Just to make sure it wasn't something incorrect on my side, I did the same with the 'e' editor. My cpu utilization rose from around ~20% to ~25% and then idled at around ~20% again.

You added the idle at the keyboard wait loop. Perhaps it is still attempting to wait for a specific amount of time to pass using a tightloop, elsewhere?

**Comment by jmalak** [comment]
Yes it is possible, I am calling INT 2Fh with AX=1680h to signal DOS the application idle time each loop cycle when it is waiting for keyboard entry.
It doesn't reduce CPU usage but signal DOS/DPMI that application is not doing something useful that CPU can be used for other job.

**Comment by osen** [comment]
I looked through your commit (wow there is a lot there, Watcom Vi sure had a bad case of bitrot ;))
It all looked good so I tried downloading from the link you gave me again (in case I downloaded it before the CI server had completed the updated build) and just tested it out now.
It seems to work great. Vi is happily idling at between 17% and 22% :)

Thanks again for your work on this. It is much appreciated.
